File: effecientcal.go
package main
import ("fmt")

func main() {
  var num int = 0
  var num2 int = 0
  var oprator string = ""
  fmt.Println("Enter number 1:")
  fmt.Scan(&num)
  fmt.Println("Enter number 2:")
  fmt.Scan(&num2)
  fmt.Println("Enter Oprator")
  fmt.Scan(&oprator)
  if oprator == "+" {
	fmt.Println(num+num2)
  }
  if oprator == "-" {
	fmt.Println(num-num2)
  }
  if oprator == "*" {
	fmt.Println(num*num2)
  }
  if oprator == "/" {
        if num2 == 0 {
            fmt.Println("Cannot divide by zero")
        } else {
            fmt.Println(num / num2)
        }
	}
}

File: effecientcal.lua
#!/usr/bin/env lua
io.write("Enter first number: ")
local num = tonumber(io.read())
io.write("Enter operation (+, -, *, /): ")
local op = io.read()
io.write("Enter second number: ")
local num2 = tonumber(io.read())

local function add(a,b)
    return a + b end
local function sub(a,b)
    return a - b end
local function mul(a,b)
    return a * b end
local function div(a,b)
    return a / b end

if op == "+" then
    print(add(num, num2))
elseif op == "-" then
    print(sub(num, num2))
elseif op == "*" then
    print(mul(num, num2))
elseif op == "/" then
    if num2 == 0 then
        print("Cannot divide by zero")
    else
        print(div(num, num2)) end
else
    print("Invalid operation") end